modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvFile.java

```diff
@@ -88,7 +90,34 @@ public static CsvFile of(CharSource source, boolean headerRow) {
    */
   public static CsvFile of(CharSource source, boolean headerRow, char separator) {
     ArgChecker.notNull(source, "source");
-    ImmutableList<String> lines = Unchecked.wrap(() -> source.readLines());
+    List<String> lines = Unchecked.wrap(() -> source.readLines());
+    return create(lines, headerRow, separator);
+  }
+
+  /**
+   * Parses the specified reader as a CSV file where the separator is specified and might not be a comma.
+   * <p>
+   * This factory method takes a {@link Reader}.
+   * Callers are encouraged to use {@link CharSource} instead of {@code Reader} where possible.
```

Comment:
Is there a reason to recommend `CharSource` instead of `Reader`? It's a puzzling recommendation with no explanation. `Reader` is a JDK class familiar to pretty much any Java dev.